socs/agents/rtsp_camera/agent.py

```diff
@@ -0,0 +1,583 @@
+        if self.fake:
+            cap = FakeCamera()
+        else:
+            cap = cv2.VideoCapture(self.connection)
+        if not cap.isOpened():
+            self.log.error(f"Cannot open RTSP stream at {self.connection}")
+            return False, "Could not open RTSP stream"
+
+        # Tracking state of whether we are currently recording motion detection
+        detecting = False
+        detect_start = None
+        record_frames = int(self.record_fps * self.record_duration)
+        motion_detector = MotionDetector()
+
+        snap_count = 0
+        while self.is_streaming:
+            # Use UTC
+            timestamp = time.time()
+            data = dict()
+
+            for iframe in range(frames_per_snapshot):
+                pmgrab.sleep()
+                # Grab an image
+                _ = cap.grab()
+            success, image = cap.retrieve()
+            if not success:
+                msg = "Failed to retrieve snapshot image from stream"
+                self.log.error(msg)
+                return False, "Broken stream"
+
+            # Motion detection.  We ignore the first few snapshots and also
+            # any changes that happen while we are already recording.
+            if snap_count < 4:
+                skip = True
+            else:
+                skip = False
+            if detecting:
+                if (snap_count - detect_start) * frames_per_snapshot > record_frames:
+                    # We must have finished recording
+                    detecting = False
+                    skip = False
+                else:
+                    # We are still recording
+                    skip = True
+            if self.motion_detect and self._in_motion_time_range():
+                image, movement = motion_detector.process(image, skip=skip)
+                if movement:
+                    # Start recording
+                    detecting = True
+                    detect_start = snap_count
+                    rec_stat, rec_msg, _ = self.agent.start(
+                        "record", params={"test_mode": False}
+                    )
+                    if rec_stat != ocs.OK:
+                        self.log.error(f"Problem with motion capture: {rec_msg}")
+
+            # Save to circular buffer
+            self.img_buffer.store(image)
+
+            # Get the saved path
+            path = self.img_buffer.fetch_index(-1)[0]
+
+            # Fill data
+            data = {
+                "address": self.address,
+                "timestamp": timestamp,
+                "path": path,
+            }
+
+            # Update session.data and publish
+            session.data = data
+            self.log.debug(f"{data}")
+
+            message = {
+                "block_name": "cameras",
+                "timestamp": timestamp,
+                "data": {
+                    "address": self.address,
+                    "path": path,
+                },
+            }
```

Comment:
Can you add `movement` bool to this? We may want to monitor on Grafana and trigger alarms if the cameras see movement.